src/libxively/tls/xi_tls_layer.c

```diff
@@ -275,18 +280,6 @@ static xi_state_t recv_handler( void* context, void* data, xi_state_t in_out_sta
         return XI_STATE_OK;
     }
 
-    /* if we suppose to reuse the buffer */
-    if ( XI_STATE_WANT_READ == in_out_state && NULL != data_desc )
-    {
```

Comment:
aha, so cross-question:
so the TLS lib's _pending_ function (e.g. wolfssl_pending) returns 0< in two cases:
- there are further TLS packages unprocessed (this is the case you found and my bugfix primarily targets)
- the current TLS package is not fully read